File: strings/prob242.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        l1 = [0] * 26
        l2 = [0] * 26
        
        offset = ord('a')
        for c in s:    
            l1[ord(c)-offset] += 1
        for c in t:
            l2[ord(c)-offset] += 1
        return l1 == l2

class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        d1 = {}
        d2 = {}
        for c in s:
            d1[c] = d1.get(c, 0) + 1
        for c in t:
            d2[c] = d2.get(c, 0) + 1

        return d1 == d2

File: hashtable/prob242.kt
/* 242 valid anagram*/
class Solution {
    fun isAnagram(s: String, t: String): Boolean {

        val n = s.length
        var m = t.length
        if (m != n) return false

        val helper = fun(s: String): MutableMap<Char, Int> {

            val m = mutableMapOf<Char, Int>()
            for (c in s) {
                m[c] = m.getOrDefault(c, 0)+1
            }
            return m
        }
        val m1 = helper(s)
        var m2 = helper(t)
        return m1 == m2

    }
}
